qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -191,18 +181,25 @@ def quantum_instance(self) -> QuantumInstance:
         return self._quantum_instance
 
     @quantum_instance.setter
-    def quantum_instance(self, quantum_instance) -> None:
+    def quantum_instance(
+        self, quantum_instance: Optional[Union[QuantumInstance, BaseBackend, Backend]]
+    ) -> None:
         """Sets the quantum instance to evaluate the circuit and make sure circuit has
         measurements or not depending on the type of backend used.
         """
         self._quantum_instance = quantum_instance
-
-        # add measurements in case non are given
-        if quantum_instance.is_statevector:
-            if len(self._circuit.clbits) > 0:
-                self._circuit.remove_final_measurements()
-        elif len(self._circuit.clbits) == 0:
-            self._circuit.measure_all()
+        if quantum_instance is not None:
+            if isinstance(quantum_instance, (BaseBackend, Backend)):
+                quantum_instance = QuantumInstance(quantum_instance)
```

Comment:
He, if we have a backend, as getter returns just a QuantumInstance type we should store the one we create here in self._quantum_instance so the getter can pass it back. Eg. see this https://github.com/Qiskit/qiskit-terra/blob/d2098068d31f4c04e7dee228a27951a3dde3b4f9/qiskit/algorithms/variational_algorithm.py#L97-L103